File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        numSet = set(nums)
        output = 0
        i = 0
        while(i < len(nums)):
            localCount = 1
            val = nums[i]
            while(val + 1 in numSet):
                localCount += 1
                val += 1
                print(nums[i], localCount)
            output = max(localCount, output)
            i += 1
        return output

File: Data Structures & Algorithms/combination-target-sum/submission-5.java
class Solution {
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        
        List<List<Integer>> ans = new ArrayList<>();
        backtrack(nums, 0, new ArrayList(), ans, target);
        return ans;
    }

    private void backtrack(int[] nums, int index, List<Integer> subset, List<List<Integer>> output, int target) {

        if (target == 0) {
            output.add(new ArrayList<>(subset));
        }

        if (target < 0) {
            return;
        }

        for(int i = index; i < nums.length; i++) {
            subset.add(nums[i]);
            backtrack(nums, i, subset, output, target - nums[i]);
            subset.removeLast();
        }

    }
}
